### PR TITLE
ellaborate on /consumer/.../records timeout parameter

### PR DESCRIPTION
**Pull request by mente**
As discussed in #481 . @stanislavkozlovski please checkout whether my understanding of the current long polling logic is correct

### CODE REVIEW

**Comment by mente** [comment]
[clabot:check]

File: docs/api.rst [hunk 1]
```diff
@@ -1307,7 +1307,7 @@ any consumers before it is terminated.
    :param string group_name: The name of the consumer group
    :param string instance: The ID of the consumer instance
 
-   :query timeout: The number of milliseconds for the underlying client library poll(timeout) request to fetch the records. Default to 5000ms.
+   :query timeout: The number of milliseconds for the underlying client library poll(timeout) request to fetch the records. Default is undefined. This parameter is used only if it's smaller than `consumer.request.timeout.ms` defined during consumer instance creation. If you didn't define any during consumer instance creation - it is ignored.
```
Comment (stanislavkozlovski) [inline docs/api.rst]:
```
If you didn't define any during consumer instance creation - it is ignored.
```
If you didn't define any during consumer instance creation, the one in the proxy config is taken (or its default value)

**Comment by mente** [comment]
Updated

File: docs/api.rst [hunk 2]
```diff
@@ -772,7 +772,7 @@ any consumers before it is terminated.
    :<json string consumer.request.timeout.ms: Sets the ``consumer.request.timeout.ms`` setting for this consumer specifically.
                                               This setting controls the maximum total time to wait for messages for a request
                                               if the maximum request size has not yet been reached.
-                                              It does not affect the underlying consumer->broker connection  
+                                              It does not affect the underlying consumer->broker connection. Default value is taken from rest proxy config file
```
Comment (stanislavkozlovski) [inline docs/api.rst]: `from rest proxy` => from "*the* *REST* proxy"

File: docs/api.rst [hunk 3]
```diff
@@ -1307,7 +1307,7 @@ any consumers before it is terminated.
    :param string group_name: The name of the consumer group
    :param string instance: The ID of the consumer instance
 
-   :query timeout: The number of milliseconds for the underlying client library poll(timeout) request to fetch the records. Default to 5000ms.
+   :query timeout: The number of milliseconds for the underlying client library poll(timeout) request to fetch the records. Default is undefined. This parameter is used only if it's smaller than the `consumer.timeout.ms` that is defined either during consumer instance creation or in the proxy's config file.
```
Comment (stanislavkozlovski) [inline docs/api.rst]: Sorry about noticing this late, but the original description is inaccurate as well. Do you mind if we change it while we're at it?

`The number of milliseconds for the underlying client library poll(timeout) request to fetch the records.` => `The maximum number of milliseconds that the proxy will spend fetching records`

Comment (mente) [reply docs/api.rst]: WDYT?

File: docs/api.rst [hunk 4]
```diff
@@ -1307,7 +1307,7 @@ any consumers before it is terminated.
    :param string group_name: The name of the consumer group
    :param string instance: The ID of the consumer instance
 
-   :query timeout: The number of milliseconds for the underlying client library poll(timeout) request to fetch the records. Default to 5000ms.
+   :query timeout: Maximum amount of milliseconds the REST proxy will spend fetching records. Other parameters controlling actual time spent fetching records: `max_bytes` and `fetch.min_bytes`. Default value is undefined. This parameter is used only if it's smaller than the `consumer.timeout.ms` that is defined either during consumer instance creation or in the proxy's config file.
```
Comment (stanislavkozlovski) [inline docs/api.rst]: LGTM, just one last minor thing - `fetch.min_bytes` => `fetch.min.bytes`

Comment (mente) [reply docs/api.rst]: done

**Comment by stanislavkozlovski** [comment]
LGTM! Thanks for the contribution, @mente!

**Pull request merged**